pkg/informer/informer_test.go

```diff
@@ -126,6 +126,50 @@ func Test_listAndResync_syncWithPrevState(t *testing.T) {
 	if *deleteFuncCount != 1 {
 		t.Errorf("Expected informer.DeleteFunc to be called 1 time, but got %d.", *deleteFuncCount)
 	}
+
+}
+
+func Test_StoppedInformer_ValidateDeleteFunc(t *testing.T) {
+	var uid string
+
+	//create informer for mock resource
+	informer, _, _, _ := initInformer()
+
+	//mock informer state
+	informer.resourceIndex["id-999"] = "12345"
+	informer.resourceIndex["id-100"] = "54321"
+
+	//keep track of calls made to mock informer.DeleteFunc below
+	deleteFuncCalls := make([]string, 0)
+
+	// mock DeleteFunc and get uids
+	informer.DeleteFunc = func(obj interface{}) {
+		for key := range informer.resourceIndex {
+			obj := newUnstructured(informer.gvr.Resource, key)
+			uid = string(obj.GetUID())
+			deleteFuncCalls = append(deleteFuncCalls, uid)
+		}
+	}
+
+	// start informer
+	stopper := make(chan struct{})
+	go informer.Run(stopper)
+	time.Sleep(10 * time.Millisecond)
+
+	//exist informer to trigger DeleteFunc
+	close(stopper)
+
+	//allow test to process
+	time.Sleep(10 * time.Millisecond)
+
+	// Verify that the informer.DeleteFunc was called with uid=id-999 and uid=id-100
+	valid := map[string]bool{"id-999": true, "id-100": true}
```

Comment:
This should match the `inform.resourceIndex`.  So you can:
1. Change to `map[string]string`
2. declare this object at the beginning of the test and use it to create the mock.
  ```
  resources := map[string]string{"id-999": "12345", "id-100": "54321"}
  //
  //
  informer.resourceIndex = resources
  ```